### PR TITLE
Add a top level panic handler

### PR DESCRIPTION
**Pull request by caglar10ur**

### CODE REVIEW

File: cmd/tether/main_darwin.go [hunk 1]
```diff
@@ -26,7 +27,12 @@ import (
 )
 
 func main() {
-	defer halt()
+	defer func() {
```
Comment (fdawg4l) [inline cmd/tether/main_darwin.go]: Do `defer` instructions get run when something PANICs?  Also, logrus has a panic handler.  See [here](https://github.com/deferpanic/dp-logrus) for an example.

Comment (caglar10ur) [inline cmd/tether/main_darwin.go]: defer is the only way to catch a panic http://devs.cloudimmunity.com/gotchas-and-common-mistakes-in-go-golang/index.html#panic_recover

Not sure why we want logrus middleware?

Comment (caglar10ur) [inline cmd/tether/main_darwin.go]: Oh I see what you are saying. Yeah not sure if it catches a panic from a function that's called from main

Comment (caglar10ur) [inline cmd/tether/main_darwin.go]: https://play.golang.org/p/45Ctcn3cgb unless I misunderstood what you are asking

**Comment by fdawg4l** [comment]
LGTM.  Good work.  Seems useful!

**Pull request merged**